core/src/blocks/llm.rs

```diff
@@ -585,11 +591,11 @@ mod tests {
                 blocks: HashMap::new(),
             },
             state: serde_json::from_str(
-                r#"{"RETRIEVE":[
+                r#"{"RETRIEVE": {"val": [
```

Comment:
As a result this should be left untouched (otherwise this would break all existing apps as they expect to see result values in the env.state`